`make lint` broken by numpy 2.5.0: mypy fails to parse numpy stubs with `python_version = 3.10`

<!--
Copyright Contributors to the Pyro project.

SPDX-License-Identifier: Apache-2.0
-->

### Issue Description

`make lint` fails on a clean checkout of `dev`. No code changes needed, it breaks
with the current unpinned dependencies:

```
mypy --install-types --non-interactive --warn-unused-ignores pyro scripts tests
.../site-packages/numpy/__init__.pyi:737: error: Type statement is only supported in Python 3.12 and greater  [syntax]
Found 1 error in 1 file (errors prevented further checking)
make: *** [Makefile:24: lint] Error 2
```

**Root cause:** numpy 2.5.0 (released 2026-06-21) dropped Python 3.11 support and
rewrote its bundled stubs to use PEP 695 `type X = ...` aliases.
Pyro's mypy config in `setup.cfg` sets `python_version = 3.10`, and mypy applies
target-version syntax checks to dependency stubs too, so it refuses to parse
`numpy/__init__.pyi`. The syntax error is blocking, so nothing gets type-checked.
Any CI run that resolves numpy >= 2.5.0 fails; last runs with numpy <= 2.4.x passed.

Related mypy feature request (syntax checks shouldn't block when running on 3.12+):
https://github.com/python/mypy/issues/21178

**Possible fixes:**
1. Bump `python_version = 3.12` in the `[mypy]` section of `setup.cfg`. Downside: mypy no longer
   validates 3.10/3.11 compatibility.
2. Pin `numpy<2.5` in the lint/dev requirements until the mypy issue above is resolved.


### Environment

- GitHub Actions (ubuntu), Python 3.12.13
- numpy 2.5.0, latest mypy
- Pyro `dev` branch

### Code Snippet

Minimal repro, independent of pyro's code:

```bash
echo "import numpy" > repro.py

uvx --python 3.12 --with "numpy>=2.5" mypy --python-version 3.10 repro.py
# numpy/__init__.pyi:737: error: Type statement is only supported in Python 3.12 and greater  [syntax]

uvx --python 3.12 --with "numpy>=2.5" mypy --python-version 3.12 repro.py
# Success: no issues found
```

`--python-version 3.10` mirrors `python_version = 3.10` in pyro's `setup.cfg`.

--- a/setup.cfg
+++ b/setup.cfg
@@ -28,7 +28,7 @@ column_limit = 120
 # Global options:
 
 [mypy]
-python_version = 3.10
+python_version = 3.12
 explicit_package_bases = True
 warn_return_any = True
 warn_unused_configs = True

